src/utils/git.js

```diff
@@ -81,23 +91,26 @@ const execGit = args => {
     let dataString = '';
     let errorString = '';
 
-    const gitResponse = spawn('git', args);
+    const gitResponse = spawn('git', args, { detached: true });
+    children.push(gitResponse);
 
     gitResponse.stdout.setEncoding('utf8');
     gitResponse.stderr.setEncoding('utf8');
 
     gitResponse.stdout.on('data', data => (dataString += data));
     gitResponse.stderr.on('data', data => (errorString += data));
 
-    gitResponse.on('close', code => {
+    gitResponse.on('close', (code, signal) => {
       if (code === 0) {
         resolve(dataString.toString());
+      } else if (signal === 'SIGTERM') {
+        reject(signal);
```

Comment:
I think this is related to the first comment you made--

process.kill(processList[0].pid) kills all processes with that pid (in case any future spawned processes have their own child processes). It appears that the processes spawned by execGit have the same pid as cio, so killing the spawned process also kills the main.

I had detached the spawned processes for this reason, but that appears to not have worked... My latest commit will have fixed that.